src/gui/mouse-event-handler.ts

```diff
@@ -203,75 +299,200 @@ export class MouseEventHandler implements IDestroyable {
 			// and try to process the drag event
 			// else we shouldn't prevent default of the event and ignore processing the drag event
 			if (!isVertDrag && !isHorzDrag) {
-				this._preventDragProcess = true;
+				this._preventTouchDragProcess = true;
 			}
+
+			this._touchMoveExceededManhattanDistance = true;
+			// if manhattan distance is more that 5 - we should cancel tap event
+			this._cancelTap = true;
+			this._clearLongTapTimeout();
+			this._resetTapTimeout();
+		}
+
+		if (!this._preventTouchDragProcess) {
+			const compatEvent = this._makeCompatEvent(moveEvent, touch);
+			this._processTouchEvent(compatEvent, this._handler.touchMoveEvent);
+
+			// we should prevent default in case of touch only
+			// to prevent scroll of the page
+			preventDefault(moveEvent);
+		}
+	}
+
+	private _mouseMoveWithDownHandler(moveEvent: MouseEvent): void {
+		if (moveEvent.button !== MouseEventButton.Left) {
+			return;
 		}
 
-		if (moveExceededManhattanDistance) {
-			this._moveExceededManhattanDistance = true;
+		const moveInfo = this._touchMouseMoveWithDownInfo(getPosition(moveEvent), ensureNotNull(this._mouseMoveStartPosition));
+		const { manhattanDistance } = moveInfo;
 
+		if (manhattanDistance >= Constants.CancelClickManhattanDistance) {
 			// if manhattan distance is more that 5 - we should cancel click event
 			this._cancelClick = true;
+			this._resetClickTimeout();
+		}
+
+		if (this._cancelClick) {
+			// if this._cancelClick is true, that means that minimum manhattan distance is already exceeded
+			const compatEvent = this._makeCompatEvent(moveEvent);
+			this._processMouseEvent(compatEvent, this._handler.pressedMouseMoveEvent);
+		}
+	}
+
+	private _touchMouseMoveWithDownInfo(currentPosition: Position, startPosition: Position): TouchMouseMoveWithDownInfo {
+		const xOffset = Math.abs(startPosition.x - currentPosition.x);
+		const yOffset = Math.abs(startPosition.y - currentPosition.y);
 
-			if (isTouch) {
-				this._clearLongTapTimeout();
+		const manhattanDistance = xOffset + yOffset;
+
+		return {
+			xOffset: xOffset,
+			yOffset: yOffset,
+			manhattanDistance: manhattanDistance,
+		};
+	}
+
+	/**
+		* In Firefox mouse events dont't fire if the mouse position is outside of the browser's border.
+		* To prevent the mouse from hanging while pressed we're subscribing on the mouseleave event of the document element.
+		* We're subscribing on mouseleave, but this event is actually fired on mouseup outside of the browser's border.
+		*/
```

Comment:
```suggestion
	/**
	 * In Firefox mouse events dont't fire if the mouse position is outside of the browser's border.
	 * To prevent the mouse from hanging while pressed we're subscribing on the mouseleave event of the document element.
	 * We're subscribing on mouseleave, but this event is actually fired on mouseup outside of the browser's border.
	 */
```

looks like tabs issues here?